### PR TITLE
Mention number of groups in performance note

### PR DESCRIPTION
**Pull request by nalimilan**
With a handful of groups, specialization does not really matter.

### CODE REVIEW

File: src/abstractdataframe/selection.jl [hunk 1]
```diff
@@ -54,9 +54,9 @@ const TRANSFORMATION_COMMON_RULES =
     6. vectors or matrices containing transformations specified by the `Pair` syntax
        described in points 2 to 5
     8. a function which will be called with a `SubDataFrame` corresponding to each group;
-       this form should be avoided due to its poor performance unless a very large
-       number of columns are processed (in which case `SubDataFrame` avoids excessive
-       compilation)
+       this form should be avoided due to its poor performance unless the number of groups
```
Comment (bkamins) [inline src/abstractdataframe/selection.jl:57]: looks good. Could you please also sync the manual entry that has the same passage. Thank you!

Comment (bkamins) [reply src/abstractdataframe/selection.jl:57]: bump

Comment (bkamins) [reply src/abstractdataframe/selection.jl:57]: bump :)

Comment (nalimilan) [reply src/abstractdataframe/selection.jl:57]: Woops. Done!

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
OK - can you please change the target to `main` from `master` before we merge?

**Pull request merged**